spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -71,9 +71,9 @@ public DataSourceReader createReader(DataSourceOptions options) {
   public DataSourceReader createReader(StructType readSchema, DataSourceOptions options) {
     Configuration conf = new Configuration(lazyBaseConf());
     Table table = getTableAndResolveHadoopConfiguration(options, conf);
-    String caseSensitive = lazySparkSession().conf().get("spark.sql.caseSensitive", "true");
+    String caseSensitive = lazySparkSession().conf().get("spark.sql.caseSensitive");
```

Comment:
Just to confirm there is a default and so there is no chance of failing with a not found exception: 

https://github.com/apache/spark/blob/7955b3962ac46b89564e0613db7bea98a1478bf2/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L279-L284